lib/manifolds/project/project.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Manifolds
+  module API
+    # Projects API
+    class Project
+      attr_reader :name, :directory
+
+      def initialize(name, directory: Pathname.new(File.join(Dir.pwd, name)))
+        self.name = name
+        self.directory = directory
+      end
+
+      def init
+        FileUtils.mkdir_p(workspaces_directory)
+        FileUtils.mkdir_p(vectors_directory)
+      end
+
+      def workspaces_directory
+        Pathname.new(directory).join("workspaces")
+      end
+
+      def vectors_directory
+        Pathname.new(directory).join("vectors")
+      end
+
+      private
+
+      attr_writer :name, :directory
```

Comment:
I'll let you and @tylr duke this one out 😄

Reply:
Intentional. I have like 300 arguments for doing it this way. But I'll leave it at it's a common pattern in most of the most popular ruby repos.

In this occurrence, I made it private because 1) I think these will get lifted to the public interface, eventually. 2)  I think there's other work that needs to happen to set the instance variable without triggering bugs. 3) I didn't have time to reason through all of that so it's effectively a placeholder method..

It's also DRYer and makes it so others don't write code needing to be refactored.

```ruby
def name=(value)
  dememoize_name!
  update_vectors!
  logger.info("name change")
  @name = value
end
```

Reply:
Bottom line. There is nowhere other than the initialiser where those instance variables are being set, so having setters is basically redundant. If there is a future need for setters, then it is super easy to introduce them. This would not be the case if there were multiple points at which these variable were being set.